addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/SpringActuatorScanRule.java

```diff
@@ -0,0 +1,193 @@
+                    String responseBody = testMsg.getResponseBody().toString();
+                    boolean matches =
+                            CONTENT_TYPE.matcher(contentType).find()
+                                    && JSON_PAYLOAD.matcher(responseBody).find();
+                    if (matches) {
+                        raiseAlert(testMsg, Alert.CONFIDENCE_MEDIUM, getRisk());
+                        break;
+                    }
+                }
+            }
+        }
+    }
+
+    private static String generatePath(String baseUriPath, String actuatorEndpoint) {
+        String newPath = "";
+        if (baseUriPath.contains("/")) {
+            if (baseUriPath.endsWith("/")) {
+                newPath = baseUriPath + actuatorEndpoint;
+            } else {
+                newPath =
+                        baseUriPath.substring(0, baseUriPath.lastIndexOf('/'))
+                                + "/"
+                                + actuatorEndpoint;
+            }
+        } else {
+            newPath = baseUriPath + "/" + actuatorEndpoint;
+        }
+        return newPath;
+    }
+
+    private HttpMessage sendActuatorRequest(String encodingType, String actuatorEndpoint) {
+        HttpMessage testMsg = getNewMsg();
+        try {
+            URI baseUri = getBaseMsg().getRequestHeader().getURI();
+            String baseUriPath = baseUri.getPath() == null ? "" : baseUri.getPath();
+            URI testUri =
+                    new URI(
+                            baseUri.getScheme(),
+                            null,
+                            baseUri.getHost(),
+                            baseUri.getPort(),
+                            generatePath(baseUriPath, actuatorEndpoint));
+            testMsg.getRequestHeader().setURI(testUri);
+            testMsg.getRequestHeader().setMethod(HttpRequestHeader.GET);
+            testMsg.getRequestHeader()
+                    .setHeader(HttpHeader.ACCEPT_ENCODING, encodingType); // Set this correctly
+            testMsg.setRequestBody("");
+            sendAndReceive(testMsg);
+            return testMsg;
+        } catch (IOException e) {
+            LOG.warn(
+                    "An error occurred while checking [{}] [{}] for {} Caught {} {}",
+                    testMsg.getRequestHeader().getMethod(),
+                    testMsg.getRequestHeader().getURI(),
+                    getName(),
+                    e.getClass().getName(),
+                    e.getMessage());
+        }
+        return null;
+    }
+
+    private void raiseAlert(HttpMessage msg, int confidence, int risk) {
+        newAlert()
+                .setRisk(risk)
+                .setConfidence(confidence)
+                .setName(getAlertName())
+                .setEvidence(msg.getResponseHeader().getPrimeHeader())
+                .setReference(getReference())
+                .setMessage(msg)
+                .setEvidence(msg.getResponseBody().toString())
```

Comment:
Is the response guaranteed to be small in all cases? If not might be better to truncate.

Reply:
Or extract the page title or some content that confirms the issue.

Reply:
hmm, gonna trim this to 100 chars. Today the response for the status page is small in JSON, once extended to heap or thread dump, could be very big.